src/repositories/sessionStore.js

```diff
@@ -0,0 +1,42 @@
+const redis = require('promise-redis')();
+
+let client;
+
+function createInstance() {
+  client = redis.createClient({
+    url: process.env.REDISCLOUD_URL,
+  });
+
+  client.on('error', (error) => {
+    /* eslint-disable-next-line no-console */
+    console.error(error);
+  });
+  return client;
+}
+
+function getInstance() {
+  if (!client) client = createInstance();
+  return client;
+}
+
+async function setSession(key, value) {
+  client = getInstance();
+  await client.set(key, JSON.stringify(value), 'EX', process.env.SESSION_EXPIRATION);
+  return key;
+}
+
+async function getSession(key) {
+  client = getInstance();
+  const result = await client.get(key);
+  if (result) {
+    return JSON.parse(result);
+  }
+  return false;
+}
+
+async function deleteSession(key) {
+  client = getInstance();
+  await client.del(key);
+}
```

Comment:
Acho que vale a pena refatorar para ser uma classe, tendo o createInstance como construtor, guardando o client do redis em `this.client`, evitando a necessidade do `getInstance`. Colocando as mesmas funções na classe, basta exportar uma instância dela e o resto do código não precisa ser alterado :)